src/packets.rs

```diff
@@ -192,6 +192,27 @@ enum PacketBufferStorage {
 }
 
 impl PacketBufferStorage {
+
+    #[inline]
+    pub fn with_capacity(capacity: usize) -> PacketBufferStorage {
+        if capacity <= INLINE_PACKET_BUFFER_SIZE {
+            PacketBufferStorage::Inline([0; INLINE_PACKET_BUFFER_SIZE / 4])
+        } else {
+            let u32_len = ((capacity - 1) / 4) + 1;
+            let mut buffer = Vec::with_capacity(u32_len);
+            unsafe { buffer.set_len(u32_len); }
```

Comment:
I'm not sure whether `set_len` is a good idea, if the vector's content is uninitialized. It's only `u32`s, so it might be okay, but I don't know. Why can't you just use the `Vec`'s capacity as the buffer's capacity and the `Vec`'s length as the actual (filled) length?

Reply:
For what I understand, it is needed in order to extract an slice from the whole vector and assign/extend the buffer easily, as all the manipulations on the data are done manually with slices [u8] rather than using the vector operations.